### ISSUE TITLE
nextcloudcmd hangs with "[ info nextcloud.sync.engine ]:      All the files are going to be changed, asking the user"

### ISSUE THREAD

**Issue by jchonig** [nextcloud/desktop#3144]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
Nextcloudcmd completes a sync.

### Actual behaviour
Nextcloudcmd sees that it is deleting the last file in a folder, issues a message: 

04-15 20:05:02:389 [ info nextcloud.sync.engine ]:      All the files are going to be changed, asking the user

and then hangs.  No matter what the setting of --non-interactive.

This is new behavior in 3.2.

### Steps to reproduce
1. Add a file to a folder
2. Sync it with nextcloudcmd
3. Delete the file on NextCloud instance
4. Run nextcloudcmd to sync it
5. Watch it hang

### Client configuration
Client version:
```
jch@nixie[600]:~$ nextcloudcmd -v
Nextcloud version 3.2.0-20210410.072919.14fbf1b10-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.2 LTS, x86_64
```

Operating system: Ubuntu focal

OS language:

Qt version used by client package (Linux only, see also Settings dialog):

Client package (From Nextcloud or distro) (Linux only):

```
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                  Version                                    Architecture Description
+++-=====================-==========================================-============-============================================================
ii  nextcloud-desktop-cmd 3.2.0-20210410.072919.14fbf1b10-1.0~focal1 amd64        folder synchronization with an Nextcloud server - cmd client
```

Installation path of client:
/usr/bin/nextcloudcmd

### Server configuration

Nextcloud version: 21.0.1

Storage backend (external storage):

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

1. Client logfile: https://gist.github.com/jchonig/ad74e517c88270bd3fe9a59b9a41b88f
2. Web server error log:

3. Server logfile: nextcloud log (data/nextcloud.log):

**Comment by jchonig** [comment]
Still an issue with 3.2.2

**Comment by leafonthewind** [comment]
Getting this same hanging behavior.  Attempted both with and without `--non-interactive`.

`06-24 14:40:03:499 [ info nextcloud.sync.engine ]:	All the files are going to be changed, asking the user`

Steps:
1. Delete or move everything out of a folder.
2. Attempt to sync it back to Nextcloud.
3. Hangs.  No files are changed on the server side.

Result of `nexcloudcmd -v`:
```
Nextcloud version 3.2.2-20210531.142805.04afaa1fe-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.2 LTS, x86_64
```

**Comment by grodrigo** [comment]
The same for me with newest version
`Nextcloud version 3.3.5-20210930.171550.0d9a6987e-1.0~focal1`
If I remove one file it syncs ok, but when removing all files from the source folder (with the exception of the .sync_ files) it hangs on 
`All the files are going to be changed, asking the user`

**Comment by clemenlg** [comment]
Same error with latest version:
Nextcloud version 3.3.5stable-Win64 (build 20210930)
Git revision 0d9a6987ebb0040b34451e51f54121dbd339ddf8
Using Qt 5.12.10, built against Qt 5.12.10
Using 'OpenSSL 1.1.1g  21 Apr 2020'
Running on Windows 10 (10.0), x86_64

**Comment by wasp38b** [comment]
I have the exact same problem!
$ nextcloudcmd -v
Nextcloud version 3.3.5-20211009.154442.0d9a6987e-1.0~focal1
Using Qt 5.15.2, built against Qt 5.12.8
Using 'OpenSSL 1.1.1k  25 Mar 2021'
Running on Debian GNU/Linux 11 (bullseye), x86_64

**Comment by Skittel** [comment]
Still a problem
--logfile and --confdir do not work either

**Comment by ehrlichja** [comment]
Same problem:

```
> nextcloud --version
Nextcloud version 3.3.5git
Git revision 0d9a6987ebb0040b34451e51f54121dbd339ddf8
Using Qt 5.15.2, built against Qt 5.15.2
Using Qt platform plugin 'xcb'
Using 'OpenSSL 1.1.1k-freebsd  25 Mar 2021'
Running on FreeBSD 13.0-RELEASE, x86_64
```

**Comment by jeritiana** [comment]
My current workaround is to have an empty file `.ncattributes` - any name will do, so that the workspace never gets empty.

My main use case is to have Nextcloud as a file transfer platform using a group folder. We usually send only one single file and the receiver deletes it when he's done copying to his local drive. So the workspace gets emptied very often. Using this dummy hidden file helps circumventing this issue.

Hope to get a fix soon!

**Comment by HyP3r-** [comment]
Problem still remains also for me. The workaround from @jeritiana does not work for me. If you take a look at the code: https://github.com/nextcloud/desktop/blob/master/src/libsync/syncengine.cpp#L750 You can see that this program is always asking for the user when files have to be removed... (`_hasRemoveFile`)

I tried to use the nextcloud client for an automated replication/synchronization which is impossible because of that.

**Comment by NoHow81** [comment]
Problem still exists

```
> nextcloudcmd -v
Nextcloud version 3.6.1stable-Win64 (build 20221018)
Git revision 5dfc0caab65080b4c3c8c3ac42bac766c11ea28c
Using Qt 5.15.2, built against Qt 5.15.2
Using 'OpenSSL 1.1.1l  24 Aug 2021'
Running on Windows 10 Version 1803, x86_64
```

**Comment by timschaeferde** [comment]
Same Problem Here:

```> nextcloudcmd -v                                                                                                         
Nextcloud version 3.5.4-20220806.084713.fea986309-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.5 LTS, x86_64
```

**Comment by serhanekicii** [comment]
Still persist.

```
Nextcloud version 3.7.3git
Git revision 5f9fa597ee8bb988eae7cab44e6dee674a510554
Using Qt 5.15.8, built against Qt 5.15.8
Using 'OpenSSL 3.0.7 1 Nov 2022'
Running on Artix Linux, x86_64
```

**Comment by dflvunoooooo** [comment]
Problem still present.
```
Nextcloud client 3.8.1
Running on Arch-Linux 
```

**Comment by mitchweaver** [comment]
Still an issue, this should be a simple bool check to see if `--non-interactive` was passed.

```
Nextcloud version 3.8.0git
Git revision b15f9376b2761b7dd9726aa3a110b1077ef57094
Running on Gentoo Linux, x86_64
```

**Comment by DaniJohni** [comment]
Problem still present.

```
#Nextcloud version  #3.7.3git
Using Qt 5.15.8, built against Qt 5.15.8
Using 'OpenSSL 3.0.8 7 Feb 2023'
Running on Debian GNU/Linux bookworm/sid, arm64
```

**Comment by BeeverTeeth** [comment]
Problem still occurs.

Nextcloud version 3.9.0-20230616.170352.c6c6c61f5-1.0~jammy1
Using Qt 5.15.3, built against Qt 5.15.3
Using 'OpenSSL 3.0.2 15 Mar 2022'
Running on Ubuntu 22.04 LTS, x86_64

**Comment by patrickuhlmann** [comment]
Will this be solved anytime? If not I might need to drop Nextcloud as it doesn't have the ability to sync unattended

**Comment by dfcsoftware** [comment]
I had the same error message leading me here, after I migrated to a new host/instance, with the same user and files.

Fixed it by removing the .sync.xxx database files in the root directory of my user.

Guessing the nextcloud instance identifiers in the database(s) are different.

**Comment by fdzaebel** [comment]
Also an issue for me. Please fix this.

**Comment by krioso** [comment]
dfcsoftware's (August 12)  answer worked for me as well - I went to the directory where I was trying to download the files to (specific user), removed the .sync_xxxxx files, nextcloudcmd was able to run after that without "asking the user".

**Comment by dataprolet** [comment]
This is still an issue and deleting the .sync file is not working every time. I'll further investigate the exact behavior and report it here.

**Comment by Knah-Tsaeb** [comment]
Problem still present.

```Nextcloud version 3.7.3git
Using Qt 5.15.8, built against Qt 5.15.8
Using 'OpenSSL 3.0.11 19 Sep 2023'
Running on Debian GNU/Linux 12 (bookworm), x86_64```

**Comment by alex-lt-kong** [comment]
Problem still exists
```
Nextcloud version 3.4.2-1ubuntu1
Using Qt 5.15.3, built against Qt 5.15.2
Using 'OpenSSL 3.0.2 15 Mar 2022'
Running on Ubuntu 22.04.4 LTS, x86_64
```

**Comment by Stinescloud** [comment]
Same Error on Year 2025

```
Nextcloud version 3.5.4-20220806.084713.fea986309-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.6 LTS, x86_64
```

**Comment by camilasan** [comment]
> Same Error on Year 2025

We just released 3.16.5, please upgrade.
